contracts/duckies/nft/DucklingFamilyGame/DucklingFamilyGameV1.sol

```diff
@@ -0,0 +1,245 @@
+
+contract DucklingFamily is IVoucher, RandomUpgradeable {
+	using CountersUpgradeable for CountersUpgradeable.Counter;
+	using Gene for uint256;
+
+	// ------- IVoucher -------
+
+	enum VouhcherActions {
+		MintPacks,
+		MeldFlocks
+	}
+
+	function useVouchers(Voucher[] calldata vouchers, bytes calldata signature) external {
+		// use vouchers to mint/meld NFTs
+	}
+
+	function useVoucher(Voucher calldata voucher, bytes calldata signature) external {
+		// use voucher to mint/meld NFTs
+	}
+
+	// ------- Ducklings Game -------
+
+	uint8 public constant MAX_PACK_SIZE = 100;
+	uint8 public constant FLOCK_SIZE = 5;
+
+	enum Collections {
+		Ducklings,
+		Zombeaks,
+		SuperLegendary
+	}
+
+	enum Classes {
+		Common,
+		Rare,
+		Epic,
+		Legendary
+	}
+
+	uint8[] internal rarityWeights; // 70, 20, 5, 1
+	uint8[] internal meldingZombeakWeights; // 10, 5, 2, 0
+
+	enum TraitDistributionTypes {
+		Even,
+		Uneven
+	}
+
+	// ------- Ducklings -------
+
+	enum DucklingTraits {
+		Collection,
+		Class,
+		Body,
+		Head,
+		Color,
+		Element,
+		Eyes,
+		Beak,
+		Wings,
+		FirstName,
+		// is the same as Element
+		// FamilyName,
+		Temper
+	}
+
+	uint8 ducklingTraitValuesNum;
+	uint32 ducklingTraitDistributionTypes;
+
+	// ------- Zombeaks -------
+
+	enum ZombeaksTraits {
+		Collection,
+		Class,
+		Body,
+		Head,
+		Eyes,
+		Beak,
+		Wings,
+		Color
+	}
+
+	uint8 zombeaksTraitValuesNum;
+	uint32 zombeaksTraitDistributionTypes;
+
+	// ------- SuperLegendary -------
+
+	enum SuperLegendaryTraits {
+		Collection,
+		Picture
+	}
+
+	uint8 superLegendaryAmount;
+
+	// ------- -------
+
+	struct Duckling {
+		uint256 gene;
+		uint64 birthdate;
+	}
+
+	string public apiBaseURL;
+	uint256 public mintPrice;
+	uint256 public meldPrice;
+
+	CountersUpgradeable.Counter public nextNewTokenId;
+	CountersUpgradeable.Counter public nextSuperLegendaryId;
+
+	mapping(uint256 => Duckling) public idToDuckling;
+
+	// ------- Initializer -------
+
+	// TODO: either pass configs as params or initialize them manually
+	function initialize() external initializer {}
+
+	// ------- Mint -------
+
+	function mintPack(uint8 size) external UseRandom {
+		// duckiesContract.burnFrom(msg.sender, mintPrice * size);
+		// _mintPackTo(msg.sender, size);
+	}
+
+	function _mintPackTo(address to, uint8 amount) internal {
+		// if (amount > MAX_PACK_SIZE) revert MintingRulesViolated(collectionId, amount);
+		// for (uint256 i = 0; i < amount; i++) {
+		// 	(uint256 tokenId, uint256 gene) = _mintTo(to, collectionId);
+		// 	emit Minted(tokenId, gene, msg.sender, block.chainid);
+		// }
+	}
+
+	function _mintTo(address to) internal returns (uint256 tokenId, uint256 gene) {
+		// gene = _generateGene(Collections.Duckling);
+		// tokenId = nextNewTokenId.current();
+		// tokenIdToDuckling[tokenId] = Duckling(gene, uint64(block.timestamp));
+		// _safeMint(to, tokenId);
+		// nextNewTokenId.increment();
+	}
+
+	function _generateGene(Collections collection) internal returns (uint256) {
```

Comment:
A gene is a specific sequence of DNA that codes for a particular protein or functional RNA molecule. In other words, a gene is a unit of heredity that is responsible for the expression of a particular trait.

A genome, on the other hand, is the complete set of genetic material of an organism. It includes all of the DNA (or RNA in some cases) in the cell, including both coding and non-coding regions. The genome contains all of an organism's genes, as well as regulatory sequences, repetitive DNA, and other non-coding regions.

Reply:
Renamed everywhere in the files: f1e7948f00ec9ab95587b20e61774e67a8048737